### PR TITLE
mac arm build and CI addition

### PR DESCRIPTION
**Pull request by mayantaylor**
Updating setup.py for correct mac build:
- charm++ build needs the architecture specified as arm8 (undoing incorrect change in #254)
- clang installation for charm4py requires architecture specified as arm64
- charmlib.so must be replaced with the mac version (.dylib)

### CODE REVIEW

**Review by matthiasdiener — DISMISSED** [review]
This looks fine, but could you add more comprehensive architecture testing to CI (either in this PR or a new one)? Currently, only linux-x86_64 seems to be tested.

File: .github/workflows/charm4py.yml [hunk 1]
```diff
@@ -28,3 +28,44 @@ jobs:
...
+        export CHARM_BUILD_PROCESSES=2
+        export CHARM4PY_BUILD_CFFI=1
+        python3 setup.py build_ext --inplace
+    - name: test-charm4py
+      run: |
+        export PYTHONPATH="$PWD"
+        export CHARM4PY_TEST_NUM_PROCESSES=2
+        python3 auto_test.py
+
+  macos-arm:
```
Comment (matthiasdiener) [inline .github/workflows/charm4py.yml:52]: is this x86_64 by any chance?

Comment (matthiasdiener) [reply .github/workflows/charm4py.yml]: Also, if the code for the CI steps is identical, this could probably be a build matrix.

**Comment by ritvikrao** [comment]
I got both the x86 and m1 mac tests to work, but i had to disable thread fork safety checking for it to do so.

**Comment by ritvikrao** [comment]
Why did you restructure it as a matrix? I split it up because I needed separate run steps for the mac build

**Comment by matthiasdiener** [comment]
> Why did you restructure it as a matrix? I split it up because I needed separate run steps for the mac build

Most of the steps are identical across the different runs; if there are small differences between the steps, we can resolve them in the build script (via e.g. `if [[ ${{ runner.os }} == 'macos-14' ]]; then ...`)

**Comment by matthiasdiener** [comment]
I think this is ready for review @mayantaylor @ritvikrao . I'm not sure if we should add other Python versions or architectures to CI?

**Comment by ritvikrao** [comment]
i think the tests here are sufficient, this looks good to me

**Review by ritvikrao — APPROVED** [review]

**Pull request merged**